### PR TITLE
[portstat]:add packet rate and port state in portstat

### PR DESCRIPTION
**Pull request by sihuihan88**
Signed-off-by: Sihui Han <sihan@microsoft.com>

### CODE REVIEW

File: scripts/portstat [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#! /usr/bin/python
```
Comment (stcheng) [inline scripts/portstat]: keep the usr/bin/env

File: scripts/portstat [hunk 2]
```diff
@@ -27,8 +27,8 @@ PORT_RATE = 40
 
 NStats = namedtuple("NStats", "rx_ok, rx_err, rx_drop, rx_ovr, tx_ok,\
                     tx_err, tx_drop, tx_ovr, rx_byt, tx_byt")
-header = ['Iface', 'RX_OK', 'RX_RATE', 'RX_UTIL', 'RX_ERR', 'RX_DRP', 'RX_OVR',
-          'TX_OK', 'TX_RATE', 'TX_UTIL', 'TX_ERR', 'TX_DRP', 'TX_OVR']
+header = ['Iface','STATE', 'RX_OK', 'RX_BRATE', 'Rx_PRATE', 'RX_UTIL', 'RX_ERR', 'RX_DRP', 'RX_OVR',
+          'TX_OK', 'TX_BRATE', 'Tx_PRATE', 'TX_UTIL', 'TX_ERR', 'TX_DRP', 'TX_OVR']
```
Comment (stcheng) [inline scripts/portstat]: capitalize to RX_PRATE and TX_PRATE

File: scripts/portstat [hunk 3]
```diff
@@ -160,6 +179,17 @@ class Portstat(object):
...
+            """
+                Calculate the packet rate.
+            """
+            if newstr == "N/A" or oldstr == "N/A":
+                return "N/A"
+            else:
+                rate = int(ns_diff(newstr, oldstr).replace(',',''))/delta
+                return "{:.2f}".format(rate)+'/s'
+
+
```
Comment (stcheng) [inline scripts/portstat]: one empty line?

File: scripts/portstat [hunk 4]
```diff
@@ -326,6 +360,7 @@ Examples:
     else:
         #wait for the specified time and then gather the new stats and output the difference.
         time.sleep(wait_time_in_seconds)
+	print "The byte rate and packet rate are calculated within %s seconds period" % wait_time_in_seconds
```
Comment (stcheng) [inline scripts/portstat]: use space instead of tabs on all places

File: scripts/portstat [hunk 5]
```diff
@@ -78,10 +81,24 @@ class Portstat(object):
...
-
+            cnstat_dict[port] = get_counters(counter_port_name_map[port]) 
         return cnstat_dict
 
+    def get_port_status(self, port_name):
+        """
+            Get the port status
+        """
+        full_table_id = PORT_STATUS_TABLE_PREFIX + port_name
+	status = self.db.get(self.db.APPL_DB, full_table_id, PORT_STATUS_FIELD)
```
Comment (lguohan) [inline scripts/portstat]: >	 [](start = 0, length = 1)

convert a tab to 4 spaces

File: scripts/portstat [hunk 6]
```diff
@@ -105,8 +122,10 @@ class Portstat(object):
                 header[9] : line[9],
                 header[10] : line[10],
                 header[11] : line[11],
-                header[12] : line[12]
-                }
+                header[12] : line[12],
+                header[13] : line[13],
+		header[14] : line[14]
```
Comment (lguohan) [inline scripts/portstat]: >		 [](start = 0, length = 2)

tab -> ws

File: scripts/portstat [hunk 7]
```diff
@@ -27,8 +27,8 @@ PORT_RATE = 40
 
 NStats = namedtuple("NStats", "rx_ok, rx_err, rx_drop, rx_ovr, tx_ok,\
                     tx_err, tx_drop, tx_ovr, rx_byt, tx_byt")
-header = ['Iface', 'RX_OK', 'RX_RATE', 'RX_UTIL', 'RX_ERR', 'RX_DRP', 'RX_OVR',
-          'TX_OK', 'TX_RATE', 'TX_UTIL', 'TX_ERR', 'TX_DRP', 'TX_OVR']
+header = ['Iface','STATE', 'RX_OK', 'RX_BRATE', 'Rx_PRATE', 'RX_UTIL', 'RX_ERR', 'RX_DRP', 'RX_OVR',
```
Comment (lguohan) [inline scripts/portstat]: let's use the same convention. Rx -> RX

let's use RX_PPS for packet per second. RX_BPS for bytes per seconds.

File: scripts/portstat (hunk 2 above)
Comment (lguohan) [inline scripts/portstat]: we are adding three lines, I really like to limit one row to one line.

by default, let's use the only show RX_BPS, then let's add option to show RX_PPS, or both.

File: scripts/portstat (hunk 7 above)
Comment (lguohan) [inline scripts/portstat]: >face','STA [](start = 12, length = 10)

make a whitespace here.

**Pull request merged**